frontend/app/services/subscription-service.server.ts

```diff
@@ -75,6 +81,55 @@ export const getSubscriptionService = moize(createSubscriptionService, { onCache
 function createSubscriptionService() {
   const { CDCP_API_BASE_URI } = getEnv();
 
+  /**
+   *
+   * @param userId
+   * @returns the subscription details for the user or null if no user is found or the user has no CDCP subscriptions.
+   */
+  async function updateUser(userId: string, userInfo: UserInfo) {
+    const auditService = getAuditService();
+    const instrumentationService = getInstrumentationService();
+    auditService.audit('alert-subscription.get', { userId });
+    const url = new URL(`${CDCP_API_BASE_URI}/api/v1/users?raoidcUserId=${userId}`);
+    const response = await fetch(url, {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    });
+
+    instrumentationService.countHttpStatus('http.client.cdcp-api.users.gets', response.status);
+
+    if (!response.ok) {
+      log.error('%j', {
+        message: 'Failed find data',
+        status: response.status,
+        statusText: response.statusText,
+        url: url,
+        responseBody: await response.text(),
+      });
+
+      throw new Error(`Failed to find data. Status: ${response.status}, Status Text: ${response.statusText}`);
+    }
+
+    const users = await response.json();
+    if (users._embedded.users.length === 0) {
+      return null;
+    }
```

Comment:
This should throw. If we attempt to update a user and that user is not found, then that should be considered an error.